Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -255,78 +260,8 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
-            linkChannelIfNeeded(channelAddedEvent.channel)
-        } else if let messageNewEvent = event as? MessageNewEvent {
-            linkChannelIfNeeded(messageNewEvent.channel)
-        } else if let messageNewEvent = event as? NotificationMessageNewEvent {
-            linkChannelIfNeeded(messageNewEvent.channel)
-        } else if let updatedChannelEvent = event as? ChannelUpdatedEvent {
-            unlinkChannelIfNeeded(updatedChannelEvent.channel)
-        } else if let channelVisibleEvent = event as? ChannelVisibleEvent, let channel = dataStore.channel(cid: channelVisibleEvent.cid) {
-            linkChannelIfNeeded(channel)
-        }
-    }
-
-    /// Handles if a channel should be linked to the current query or not.
-    private func linkChannelIfNeeded(_ channel: ChatChannel) {
-        guard !channels.contains(channel) else { return }
-        guard shouldChannelBelongToCurrentQuery(channel) else { return }
-        link(channel: channel)
-    }
-
-    /// Handles if a channel should be unlinked from the current query or not.
-    private func unlinkChannelIfNeeded(_ channel: ChatChannel) {
-        guard channels.contains(channel) else { return }
-        guard !shouldChannelBelongToCurrentQuery(channel) else { return }
-        worker.unlink(channel: channel, with: query)
-    }
-
-    /// Checks if the given channel should belong to the current query or not.
-    private func shouldChannelBelongToCurrentQuery(_ channel: ChatChannel) -> Bool {
-        if let filter = filter {
-            return filter(channel)
-        }
-
-        guard !client.config.isChannelAutomaticFilteringEnabled else {
-            // When auto-filtering is enabled the channel will appear or not automatically if the
-            // query matches the DB Predicate. So here we default to saying it always belong to the current query.
-            return true
-        }
-
-        // Fallback to legacy of checking if a channel should be linked to the current channel list query.
-        let deprecatedFallback: () -> Bool? = {
-            self.delegate?.controller(self, shouldAddNewChannelToList: channel)
-        }
-
-        return deprecatedFallback() ?? true
-    }
-
-    /// Links the channel to the current channel list query and starts watching it.
-    private func link(channel: ChatChannel) {
-        worker.link(channel: channel, with: query) { [weak self] error in
-            if let error = error {
-                log.error(error)
-                return
-            }
-
-            self?.worker.startWatchingChannels(withIds: [channel.cid]) { error in
-                guard let error = error else { return }
-                log.warning(
-                    "Failed to start watching linked channel: \(channel.cid), error: \(error.localizedDescription)"
-                )
-            }
-        }
-    }
+    public func eventsController(_ controller: EventsController, didReceiveEvent event: Event) {}
```

Comment:
do we still need this?

Reply:
No, but since it is public, I did not touch. If OK, I can throw it away.

Reply:
Do you think it is fine?

Reply:
Yes we can removove this 👍 It is public, but not overridable. I think it was public because of the requirements of the Delegate being public. But no one will really call this function.

Reply:
Removed